src/debugAdapter/runtimeInterface.ts

```diff
@@ -138,16 +136,25 @@ export default class RuntimeInterface extends EventEmitter {
    * @returns
    */
   public async attach(args: Required<DebuggerTypes.DebugArgs>): Promise<void> {
+    // Retreives the truffle configuration file
+    const config = Config.detect({workingDirectory: args.workingDirectory});
+
+    // Validate the network parameter
+    this.validateNetwork(config, args);
+
+    // Retreives the environment configuration
+    await Environment.detect(config);
+
     // Gets the contracts compilation
-    const result = await prepareContracts(args.workingDirectory);
+    const result = await prepareContracts(config);
 
     // Sets the properties to use during the debugger process
     this._mappedSources = result.mappedSources;
-    const networkId = args.disableFetchExternal ? undefined : this.getNetworkId(args.providerUrl);
+    const networkId = args.disableFetchExternal ? undefined : config.network_id;
 
     // Sets the truffle debugger options
     const options: truffleDebugger.DebuggerOptions = {
-      provider: args.providerUrl,
+      provider: config.provider.host,
```

Comment:
I believe we need to remove `host`. `config.provider` is already a Web3js provider, so need to use `host` here. Moreover, `host` does not allow us to use providers using `HDWalletProvider`.